src/components/DisplayToy.js

```diff
@@ -4,29 +4,32 @@ import { Container, Button } from 'react-bootstrap';
 import { useDispatch, useSelector } from 'react-redux';
 import { deleteToy } from '../actions/deleteToy.js';
 import { selectToy } from '../selectors/toySelectors';
-import { useHistory } from 'react-router';
+import { useParams } from 'react-router-dom';
+import { Redirect } from 'react-router-dom';
 
-const DisplayToy = (props) => {
+const DisplayToy = () => {
   const dispatch = useDispatch();
+  const { id } = useParams();
 
-  const toy = useSelector(selectToy(props.match.params.id));
+  const toy = useSelector(selectToy(id));
 
-  const history = useHistory();
-
-  return (
-    <Container>
-      <div>
-        <h1 className="header text-center">{toy.name}</h1>
-        <img src={toy.image_url} alt="toy-image" width="300" height="300" />
-        <p>{toy.description}</p>
-        <Button onClick={() => dispatch(deleteToy(toy.id, history))}>
-          Delete Toy
-        </Button>
-        <ReviewsContainer toy={toy} />
-      </div>
-      )
-    </Container>
-  );
+  if (toy)
+    return (
+      <Container>
+        <div>
+          <h1 className="header text-center">{toy.name}</h1>
+          <img src={toy.image_url} alt="toy-image" width="300" height="300" />
+          <p>{toy.description}</p>
+          <Button onClick={() => dispatch(deleteToy(toy.id))}>
+            Delete Toy
+          </Button>
+          <ReviewsContainer toy={toy} />
+        </div>
+      </Container>
+    );
+  else {
+    return <Redirect to="/toys" />;
+  }
```

Comment:
This is personal preference, but if I have a long `if` block and else is just a little one-liner, I find it easier to read if you short circuit earlier.  For example:

```
if (!toy) return <Redirect to="/toys" />;

return(
  <Container>
    ...
  </Container>
);
```

Reply:
It definitely looks better like that!